connect/src/com/telenor/connect/AbstractSdkProfile.java

```diff
@@ -96,4 +115,21 @@ public void failure(RetrofitError error) {
                     }
                 });
     }
+
+    @Override
+    public String getLastAuthState() {
+        return lastAuthState;
+    }
+
+    @Override
+    public void validateTokens(ConnectTokensTO tokens, Date serverTimestamp) {
+        Validator.notNullOrEmpty(tokens.getAccessToken(), "access_token");
```

Comment:
hmm, are u sure, did u test it with an empty scope?
doesn't json parser expects scope when parsing ConnectTokensTO ?